pyfar/classes/transmission_matrix.py

```diff
@@ -0,0 +1,670 @@
+            Per default, no additional dimensions are created.
+
+        Returns
+        -------
+        tmat : TransmissionMatrix
+            A TransmissionMatrix object that contains 2x2 identity matrices.
+
+        """
+        A = np.ones((*abcd_cshape, np.array(frequencies).size))
+        (B,C,D) = (0,0,1)
+        return tmat.from_abcd(A, B, C, D, frequencies)
+
+    @classmethod
+    def create_series_impedance(tmat, impedance: FrequencyData,
+                                abcd_cshape=()):
+        r"""Creates a transmission matrix representing a series impedance.
+
+        This means the impedance is connected in series with a potential load
+        impedance. See Equation (2-8) in Table I of Reference [1]_:
+
+        .. math::
+            T = \begin{bmatrix}
+                1 & Z \\
+                0 & 1
+                \end{bmatrix}
+
+        Parameters
+        ----------
+        impedance : FrequencyData
+            The impedance data of the series impedance.
+        abcd_cshape : tuple | int, optional
+            Shape of additional channel dimensions besides (2,2).
+            Per default, no additional dimensions are created.
+
+        Returns
+        -------
+        tmat : TransmissionMatrix
+            A TransmissionMatrix representing a series impedance.
+
+        """
+        if impedance.cshape != (1,):
+            raise ValueError("Number of channels for 'impedance' must be 1.")
+
+        (Z, frequencies) = (impedance.freq[0], impedance.frequencies)
+        Z = np.zeros((*abcd_cshape, np.array(frequencies).size)) + Z
+        return tmat.from_abcd(1, Z, 0, 1, frequencies)
+
+    @classmethod
+    def create_shunt_admittance(tmat, admittance: FrequencyData,
+                                abcd_cshape = ()):
+        r"""Creates a transmission matrix representing a shunt admittance
+        (parallel connection).
+
+        In this case, the impedance (= 1 / admittance) is connected in parallel
+        with a potential load impedance.
+        See Equation (2-9) in Table I of Reference [1]_:
+
+        .. math::
+            T = \begin{bmatrix}
+                1 & 0 \\
+                Y & 1
+                \end{bmatrix}
+
+        Parameters
+        ----------
+        admittance : FrequencyData
+            The admittance data of the element connected in parallel.
+        abcd_cshape : tuple | int, optional
+            Shape of additional channel dimensions besides (2,2).
+            Per default, no additional dimensions are created.
+
+        Returns
+        -------
+        tmat : TransmissionMatrix
+            A TransmissionMatrix representing a parallel connection.
+
+        """
+        if admittance.cshape != (1,):
+            raise ValueError("Number of channels for 'admittance' must be 1.")
+
+        (Y, frequencies) = (admittance.freq[0], admittance.frequencies)
+        Y = np.zeros((*abcd_cshape, np.array(frequencies).size)) + Y
```

Comment:
Possible to work with https://numpy.org/doc/stable/reference/generated/numpy.broadcast_to.html instead?